File: API/demo/src/main/java/com/example/demo/model/Mood.java
package com.example.demo.model;

import com.example.demo.Enum.MoodState;
import jakarta.persistence.*;

import java.time.LocalDate;

@Entity
public class Mood {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "usuario_id")
    private Users usuario;
    private LocalDate fecha;
    @Enumerated(EnumType.STRING)
    private MoodState estado;
    private String nota;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Users getUsuario() {
        return usuario;
    }

    public void setUsuario(Users usuario) {
        this.usuario = usuario;
    }

    public LocalDate getFecha() {
        return fecha;
    }

    public void setFecha(LocalDate fecha) {
        this.fecha = fecha;
    }

    public MoodState getEstado() {
        return estado;
    }

    public void setEstado(MoodState estado) {
        this.estado = estado;
    }

    public String getNota() {
        return nota;
    }

    public void setNota(String nota) {
        this.nota = nota;
    }
}

File: Aplicacion Movil/MedMonitor/app/src/main/java/Models/Mood.kt
package Models

import Enums.EstadoAnimo
import com.google.gson.annotations.SerializedName
import java.io.Serializable
import java.time.LocalDate

data class Mood(
    @SerializedName("id") val id: Long,
    @SerializedName("usuario_id") val usuarioId: Int,
    @SerializedName("fecha") val fecha: LocalDate,
    @SerializedName("estado") val estado: EstadoAnimo,
    @SerializedName("nota") val nota: String
):Serializable
